graylog2-server/src/main/java/org/graylog2/rest/resources/system/indexer/responses/IndexSetSummary.java

```diff
@@ -67,17 +69,24 @@ public static IndexSetSummary create(@JsonProperty("id") @Nullable String id,
                                          @JsonProperty(FIELD_INDEX_TEMPLATE_TYPE) @Nullable String templateType,
                                          @JsonProperty(FIELD_PROFILE_ID) @Nullable String fieldTypeProfile,
                                          @JsonProperty(FIELD_DATA_TIERING) @Nullable DataTieringConfig dataTiering,
-                                         @JsonProperty(FIELD_USE_LEGACY_ROTATION) @Nullable Boolean useLegacyRotation) {
+                                         @JsonProperty(FIELD_USE_LEGACY_ROTATION) @Nullable Boolean useLegacyRotation,
+                                         @JsonProperty(FIELD_HAS_FAILED_SNAPSHOT) Boolean hasFailedSnapshot,
+                                         @JsonProperty(FIELD_FAILED_SNAPSHOT_NAME) @Nullable String failedSnapshotName) {
         if (Objects.isNull(creationDate)) {
             creationDate = ZonedDateTime.now();
         }
         return new AutoValue_IndexSetSummary(shards, replicas, rotationStrategyClass, rotationStrategy, retentionStrategyClass,
                 retentionStrategy, dataTiering, id, title, description, isDefault, canBeDefault, isWritable, indexPrefix,
                 creationDate, indexAnalyzer, indexOptimizationMaxNumSegments, indexOptimizationDisabled, fieldTypeRefreshInterval,
-                Optional.ofNullable(templateType), fieldTypeProfile, Objects.isNull(useLegacyRotation) || useLegacyRotation);
+                Optional.ofNullable(templateType), fieldTypeProfile, Objects.isNull(useLegacyRotation) || useLegacyRotation,
+                hasFailedSnapshot, failedSnapshotName);
     }
 
     public static IndexSetSummary fromIndexSetConfig(IndexSetConfig indexSet, boolean isDefault) {
+        return fromIndexSetConfig(indexSet, isDefault, Optional.empty());
```

Comment:
If I'm not mistaken, existing calls, e.g. the API list call calls, if they still use this method, will now potentially have different values for the snapshot fields of the returned `IndexSetSummary` than the calls for an individual summary when using the new method. I think this can be misleading.

E.g. `hasFailedSnapshot == false` could mean that there is no failed snapshot but also mean that we didn't even check. 

I think it's a not ideal that the snapshot information is placed on the top level of `IndexSetSummary`. Otherwise we could easily signal, e.g. in a list context, that the snapshot information is missing by omitting the value for the top level field.

What do you think about adding an optional top level field like `dataTieringStatus` or `dataTieringInfo` (preferably something more meaningful even), and putting the failure and/or snapshot information in there? Then the absence of the top level field means that the info is not available.